### PR TITLE
Feature/natures

### PR DESCRIPTION
**Pull request by evanfuhr**

### CODE REVIEW

**Comment by evanfuhr** [comment]
It's a big one!

File: app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java [hunk 1]
```diff
@@ -95,6 +96,14 @@ public void onClick(View view) {
                 onClickButton(view);
             }
         });
+
+        _natureButton = (Button) this.findViewById(R.id.buttonNatureList);
+        _natureButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                onClickButton(view);
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java]: So am I reading this correctly? The view specific to `_natureButton` passes its view to the `onClickButton()` and process its case to deliver the intent for NatureListActivity.class?

File: app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java [hunk 2]
```diff
@@ -95,6 +96,14 @@ public void onClick(View view) {
                 onClickButton(view);
             }
         });
+
+        _natureButton = (Button) this.findViewById(R.id.buttonNatureList);
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java]: Is this casting to `(Button)` necessary?

File: app/src/main/java/com/evanfuhr/pokemondatabase/adapters/NatureRecyclerViewAdapter.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        public final View mView;
+        public final Button _button;
+        public Nature mItem;
+
+        public ViewHolder(View view) {
+            super(view);
+            mView = view;
+            _button = (Button) view.findViewById(R.id.singleButton);
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/adapters/NatureRecyclerViewAdapter.java]: Casting to `(Button)` necessary?

File: app/src/main/java/com/evanfuhr/pokemondatabase/data/FlavorDAO.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+        String sql = SQLiteQueryBuilder
+                .select(field(CONTEST_TYPES, ID)
+                        , field(CONTEST_TYPE_NAMES, NAME))
+                .from(CONTEST_TYPES)
+                .join(CONTEST_TYPE_NAMES)
+                .on(field(CONTEST_TYPES, ID) + "=" + field(CONTEST_TYPE_NAMES, CONTEST_TYPE_ID))
+                .where(field(CONTEST_TYPE_NAMES, NAME) + " LIKE LOWER('%" + nameSearchParam + "%')")
+                .and(field(CONTEST_TYPE_NAMES, LOCAL_LANGUAGE_ID) + "=" + _language_id)
+                .build();
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/data/FlavorDAO.java]: Much better!

File: app/src/main/java/com/evanfuhr/pokemondatabase/models/Flavor.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            case 2:
+                return "#9DB7F5"; // Dry    | Special Attack    | Beauty
+            case 3:
+                return "#FA92B2"; // Sweet  | Speed             | Cute
+            case 4:
+                return "#A7DB8D"; // Bitter | Special Defense   | Smart
+            case 5:
+                return "#FAE078"; // Sour   | Defense           | Tough
+            default:
+                return "#000000";
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/models/Flavor.java]: What's the significance of using integers? Are those...

oh i see, need to refactor to enum

File: app/src/main/res/layout/activity_nature_list.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        android:name="com.evanfuhr.pokemondatabase.fragments.NatureListFragment"
+        android:id="@+id/natureListFragment"
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        tools:layout_editor_absoluteY="16dp"
+        android:layout_marginRight="8dp"
+        app:layout_constraintRight_toRightOf="parent"
+        android:layout_marginLeft="8dp"
+        app:layout_constraintLeft_toLeftOf="parent" />
+
+</android.support.constraint.ConstraintLayout>
```
Comment (lolptdr) [inline app/src/main/res/layout/activity_nature_list.xml]: Keep one blank line at the end of each file.

File: app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java (hunk 1 above)
Comment (evanfuhr) [reply app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java]: Yes. The names are not particularly descriptive.

File: app/src/main/res/layout/activity_nature_list.xml (hunk 6 above)
Comment (evanfuhr) [reply app/src/main/res/layout/activity_nature_list.xml]: Ugh! Maybe I can set something in my IDE?

File: app/src/main/java/com/evanfuhr/pokemondatabase/adapters/NatureRecyclerViewAdapter.java (hunk 3 above)
Comment (evanfuhr) [reply app/src/main/java/com/evanfuhr/pokemondatabase/adapters/NatureRecyclerViewAdapter.java]: It used to be. Maybe the latest buildtools make it unnecessary? Moving forward, I'll remove as the IDE recommends.

**Comment by evanfuhr** [comment]
I'll make relevant updates and merge. Thanks!

**Pull request merged**